### PR TITLE
Implement grouped difficulty icons

### PR DESCRIPTION
**Pull request by frenzibyte**
- This fixes direct panels having out-of-bounds difficulties by replacing them with counters for each ruleset, Matching the web for now (same for carousel).
- [x] Depends on #5822

Fixes panel part of #4929

### CODE REVIEW

File: osu.Game/Beatmaps/Drawables/GroupedDifficultyIcon.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+using System.Linq;
+using osu.Framework.Graphics;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Rulesets;
+using osuTK.Graphics;
+
+namespace osu.Game.Beatmaps.Drawables
+{
+    public class GroupedDifficultyIcon : DifficultyIcon
```
Comment (peppy) [inline osu.Game/Beatmaps/Drawables/GroupedDifficultyIcon.cs]: could we please at very least xmldoc classes describing what they are when the name is not enough to do so?

Comment (iiSaLMaN) [reply osu.Game/Beatmaps/Drawables/GroupedDifficultyIcon.cs]: oh sorry, i forgot to add xmldocs

**Comment by peppy** [comment]
I've basically rewritten most pieces of this into a usable state. Please make sure to go through my commits and check the changes one by one.

**Review by peppy — APPROVED** [review]

**Comment by peppy** [comment]
Also as a side-note, this should have been split into two PRs - one for the direct overlay and one for song select.

**Pull request merged**